common/lib/dependabot/errors.rb

```diff
@@ -70,12 +73,13 @@ def self.fetcher_error_details(error)
       {
         "error-type": "octokit_rate_limited",
         "error-detail": {
-          "rate-limit-reset": error.response_headers["X-RateLimit-Reset"]
+          "rate-limit-reset": T.cast(error, Octokit::Error).response_headers["X-RateLimit-Reset"]
         }
       }
     end
   end
 
+  sig { params(error: StandardError).returns(T.nilable(T::Hash[T.any(Symbol, String), T.untyped])) }
```

Comment:
We expect keys here:

https://github.com/dependabot/dependabot-core/blob/ea7ad41e384f2e760d2b956cbd881c7afd10a35e/updater/lib/dependabot/update_files_command.rb#L99-L108

Reply:
You're right! But then I think it's actually the other way around, we're always returning symbols as keys in all these three methods, because we always use colon notation?

Reply:
I'll change the references to `parser_error_details` to use strings, instead of symbols, cause we're clearly returning strings as keys in the hash.

Reply:
```
irb(main):001> { "error-type": "foo" }.keys.first.class
=> Symbol
```

Reply:
Glad the updater specs caught it!

Reply:
🤦 Okay changing this to `Symbol`.